src/ray/common/scheduling/cluster_resource_scheduler.h

```diff
@@ -63,15 +60,59 @@ struct TaskRequest {
   std::string DebugString();
 };
 
+// Task request specifying instances for each resource.
+struct TaskResourceInstances {
```

Comment:
Borderline, but I'd prefer to make these classes

https://google.github.io/styleguide/cppguide.html#Structs_vs._Classes

Reply:
Let's discuss this in person, and maybe do it in the next PR. If classes, I'd still like to have the members public (not have another method accessing them) due to performance reasons. But then, why do a class? Again, let's talk in person.

Reply:
It's just a convention that makes it easier to understand the code without context - `struct`s should generally only be used to wrap data and not to define. This also shouldn't have any impact on performance - classes and structs are completely the same under the hood AFAIK and getters/setters will be optimized away.